ui/app/src/components/Header.tsx

```diff
@@ -35,17 +34,117 @@ import Archive from 'mdi-material-ui/Archive';
 import ShieldStar from 'mdi-material-ui/ShieldStar';
 import Menu from 'mdi-material-ui/Menu';
 import Compass from 'mdi-material-ui/Compass';
+import AccountCircle from 'mdi-material-ui/AccountCircle';
+import Logout from 'mdi-material-ui/Logout';
+import Brightness4 from 'mdi-material-ui/Brightness4';
+import Brightness5 from 'mdi-material-ui/Brightness5';
 import React, { MouseEvent, useState } from 'react';
 import { useSnackbar } from '@perses-dev/components';
 import { useProjectList } from '../model/project-client';
 import { useDarkMode } from '../context/DarkMode';
 import { useIsLaptopSize, useIsMobileSize } from '../utils/browser-size';
-import { AdminRoute, ConfigRoute, ExploreRoute, MigrateRoute } from '../model/route';
+import { AdminRoute, ConfigRoute, ExploreRoute, MigrateRoute, SignInRoute } from '../model/route';
+import { useIsAuthEnable } from '../context/Config';
+import { useAuthToken, useLogoutMutation } from '../model/auth-client';
 import WhitePersesLogo from './logo/WhitePersesLogo';
 import PersesLogoCropped from './logo/PersesLogoCropped';
 
 const ITEM_HEIGHT = 48;
 
+function ThemeSwitch(props: { isAuthEnable: boolean }) {
+  const { isDarkModeEnabled, setDarkMode } = useDarkMode();
+  const { exceptionSnackbar } = useSnackbar();
+  const handleDarkModeChange = () => {
+    try {
+      setDarkMode(!isDarkModeEnabled);
+    } catch (e) {
+      exceptionSnackbar(e);
+    }
+  };
+  const swapIcon = () => {
+    return isDarkModeEnabled ? <Brightness5 id="dark" /> : <Brightness4 id="light" />;
+  };
+  if (props.isAuthEnable) {
+    return (
+      <MenuItem onClick={handleDarkModeChange}>
+        <ListItemIcon>{swapIcon()}</ListItemIcon>
+        Theme
```

Comment:
```suggestion
        Switch Theme
```
I think it's better, more explanatory